ui/app/src/components/workspaces/WorkspaceUsers.tsx

```diff
@@ -49,104 +97,204 @@ export const WorkspaceUsers: React.FunctionComponent = () => {
           })),
         )
-        .sort((a: { role: string }, b: { role: string }) =>
-          a.role.localeCompare(b.role),
+        .sort((a: { role: any }, b: { role: any }) =>
+          a.role.id.localeCompare(b.role.id),
         );
 
       setState({ users, apiError: undefined, loadingState: LoadingState.Ok });
     } catch (err: any) {
       err.userMessage = "Error retrieving users";
       setState({ users: [], apiError: err, loadingState: LoadingState.Error });
     }
+    setloadingUsers(false);
   }, [apiCall, workspace.id, roles.length, workspaceApplicationIdURI]);
 
+  const addedAssignment = async () => {
+    navigate(-1);
+    await getUsers();
+  }
+
   useEffect(() => {
     getUsers();
   }, [getUsers]);
 
-  const groupedUsers = useMemo(() => {
-    const groups: { [key: string]: IUser[] } = {};
-    state.users.forEach((user) => {
-      if (!groups[user.role]) {
-        groups[user.role] = [];
+  // De-assign user from role
+  const deassignUser = useCallback(async () => {
+    try {
+      setDeassigning(true);
+
+      await apiCall(`${ApiEndpoint.Workspaces}/${workspace.id}/${ApiEndpoint.Users}/assign?user_id=${selectedUserRole?.user_id}&role_id=${selectedUserRole?.role.id}`,
+        HttpMethod.Delete, "");
+
+      await getUsers();
+
+
+      setSelectedUserRole(undefined);
+      setHideCancelDialog(true);
+      setDeassigning(false);
+
+    } catch (err: any) {
+      err.userMessage = 'Error deassigning user';
+      setApiError(err);
+      setDeassignmentError(true);
+      setDeassigning(false);
+    }
+  }, [apiCall, selectedUserRole, workspace.id, getUsers]);
+
+  const groups: IGroup[] = useMemo(() => {
+    const groupMap: any = {};
+    const groups: any = [];
+    let currentIndex = 0;
+
+    state.users.forEach(user => {
+      if (!groupMap[user.role.id]) {
+        groupMap[user.role.id] = {
+          count: 0,
+          key: user.role.id,
+          name: user.role.displayName,
+          startIndex: currentIndex,
+          level: 0
+        };
+
+        groups.push(groupMap[user.role.id]);
       }
-      groups[user.role].push(user);
+
+      groupMap[user.role.id].count += 1;
+      currentIndex += 1;
     });
+
     return groups;
   }, [state.users]);
 
-  const groups: IGroup[] = useMemo(() => {
-    return Object.keys(groupedUsers).map((role, index) => ({
-      key: role,
-      name: role,
-      startIndex: index,
-      count: groupedUsers[role].length,
-    }));
-  }, [groupedUsers]);
+  const selection = useMemo(() => {
+    const s = new Selection({
+      onSelectionChanged: () => {
+        setSelectedUserRole(s.getSelection()[0] as IUser);
+      }
+    });
+    s.setItems(state.users, true);
+    return s;
+  }, [state.users]);
 
-  const columns: IColumn[] = [
+  const columns: IColumn[] = useMemo(() => [
     {
       key: "name",
       name: "Name",
       fieldName: "name",
       minWidth: 150,
-      onRender: (item: User) => (
+      onRender: (item: IUser) => (
```

Comment:
I'm curious. Why the change from the type `User` to the interface `IUser`? What the original code "wrong"?

Reply:
IUser was a specific construct that contains data in the right shape for the GroupedList.  We found that when you try to add and remove items from the list it needed specific fields (e.g. key).  Arguably it needs a better name.

Reply:
@damoodamoo @jjgriff93 do you have any views on this?